docs/source/federation/migrating-from-stitching.md

```diff
@@ -3,83 +3,33 @@ title: Migrating from schema stitching
 description: How to migrate services to Apollo Federation
 ---
 
-## Comparison with schema stitching
-
-If you're familiar with the `graphql-tools` schema stitching implementation, you're likely wondering how that compares to federation. There are three major differences between the approaches:
-
-1. With federation, microservices all expose a proper part of the overall graph and can refer directly to types that live in other services, without the need to add foreign keys or superfluous relationship root fields to your schema.
-2. Federation is fully declarative and doesn't require any user code to be running in the gateway.
-3. Execution is efficient and predictable, because it relies on a query plan generated ahead of time with full knowledge of the overall query, instead of on runtime schema delegation that gets invoked as part of normal resolver-based execution.
-
-The first point may need some unpacking. Schema stitching forces you to expose foreign keys and relationship root fields at service boundaries instead of referring to external types directly.
-
-For example, your reviews service would expose  `authorID` and `productID` fields, instead of fields that link to `User` or `Product`. You're also forced to add relationship root fields to get the reviews for a user or product by ID:
-
-```graphql
-extend type Query {
-  reviewsForUser(userID: ID!): [Review]
-}
-
-type Review {
-  body: String
-  authorID: ID!
-}
-```
+## Introduction
 
-This means the relationships are implicit, and you can't create a proper graph just based on this schema. In order to compose this service into an overall graph, you'll have to add link definitions to the gateway:
+This guide is meant to support users who are already operating a data graph with a gateway built using schema-stitching, and specifically to discuss the **how** of migrating infrastructure and workflows in a way that is both incremental and backwards-compatible. This guide does **not** provide details of _why_ you may want to migrate, and we encourage you to read [the announcement post](https://blog.apollographql.com/apollo-federation-f260cf525d21) to learn some more about how we built Apollo Federation as an answer to problems teams encountered with schema-stitching, and [review the comparison between the two](#comparison-with-schema-stitching) below.
 
-```graphql
-extend type Review {
-  author: User
-}
-
-extend type User {
-  reviews: [Review]
-}
+# Preparing for a migration
 
-```
+The basic strategy for migrating from a stitching gateway to Apollo Federation is to start by standing up an Apollo gateway side-by-side with the stitching gateway, and then migrating services underneath the gateway one-by-one until the graph is completely migrated. The essential steps are these:
 
-And you'll also have to write the code that implements these links, code that will be running in the gateway:
+1. [Add basic federation support to all stitched services](#adding-federation-support-to-services)
```

Comment:
By identifying all participating services and making sure that services can support federation primitives, you will uncover source-language support for Federation.  For example, Java, Javascript, and .NET support Federation at the time of writing, but other languages may require additional work.

Reply:
.NET supports it, too? 😮